src/Reveal.Sdk.Dom.Tests/Visualizations/VisualizationFixture.cs

```diff
@@ -0,0 +1,445 @@
+            var testTitle = "testTitle";
+
+            // Act
+            var visualization = new Mock<Visualization>(testTitle, dataSourceItem) { CallBase = true }.Object;
+
+            // Assert
+            Assert.Equal(testTitle, visualization.Title);
+            Assert.Equal(ChartType.Unknown, visualization.ChartType);
+        }
+
+
+        [Fact]
+        public void Constructor_UseTabularDataDefinition_WhenUseTabularDataSource()
+        {
+            // Arrange
+            var dataSource = new DataSource();
+            var dataSourceItem = new DataSourceItem
+            {
+                DataSource = dataSource,
+                HasTabularData = true,
+                Fields = new List<IField>
+                {
+                    new TextField("A"),
+                    new TextField("B"),
+                    new TextField("C"),
+                    new TextField("D"),
+                }
+            };
+
+            // Mock join tables
+            var leftJoinFieldName = "left";
+            var rightJoinFieldName = "right";
+            var aliasName = "Alias";
+            var joinConditions = new List<JoinCondition> { new JoinCondition(leftJoinFieldName, rightJoinFieldName) };
+            var dataSourceItemToJoin = new DataSourceItem();
+
+            dataSourceItem.Join(aliasName, leftJoinFieldName, rightJoinFieldName, dataSourceItemToJoin);
+
+            // Act
+            var mock = new Mock<Visualization>("testTitle", dataSourceItem) { CallBase = true };
+            var visualization = mock.Object;
+
+            // Assert
+            Assert.IsType<TabularDataDefinition>(visualization.DataDefinition);
+            Assert.Equal(dataSourceItem, visualization.DataDefinition.DataSourceItem);
+            Assert.Equal(dataSourceItem.Fields.Count, (visualization.DataDefinition as TabularDataDefinition).Fields.Count);
+            Assert.Equal(dataSourceItem.JoinTables.Count, (visualization.DataDefinition as TabularDataDefinition).JoinTables.Count);
+        }
+
+        [Fact]
+        public void Constructor_UseXmlaDataDefinition_WhenNotUseTabularDataSource()
+        {
+            // Arrange
+            var dataSource = new DataSource();
+            var dataSourceItem = new DataSourceItem
+            {
+                DataSource = dataSource,
+                HasTabularData = false,
+            };
+
+            // Act
+            var visualization = new Mock<Visualization>("testTitle", dataSourceItem) { CallBase = true }.Object;
+
+            // Assert
+            Assert.IsType<XmlaDataDefinition>(visualization.DataDefinition);
+        }
+
+        [Fact]
+        public void Id_IsUnique_WhenConstructed()
+        {
+            // Arrange
+            var dataSource = new DataSource();
+            var dataSourceItem = new DataSourceItem
+            {
+                DataSource = dataSource,
+                HasTabularData = false,
```

Comment:
We should only set HasTabularData value here in case we want to write unit test for its specific value